pkgs/servers/nextcloud/packages/apps/recognize.nix

```diff
@@ -0,0 +1,103 @@
+{
+  stdenv,
+  fetchurl,
+  lib,
+  nodejs,
+  python311,
+  util-linux,
+  ffmpeg,
+}:
+stdenv.mkDerivation rec {
+  pname = "nextcloud-app-recognise";
+  version = "8.1.1";
+
+  srcs = [
+    (fetchurl {
+      inherit version;
+      url = "https://github.com/nextcloud/recognize/releases/download/v${version}/recognize-${version}.tar.gz";
+      hash = "sha256-RIa2HzX2hfgS7KHGGBsxEdUCqDqXeAs3Xi288qA1gU4=";
+    })
+
+    (fetchurl {
+      inherit version;
+      url = "https://github.com/nextcloud/recognize/archive/refs/tags/v${version}.tar.gz";
+      hash = "sha256-op1Fdm40d8V6V+rfne0ECr60xRjBYrBPrCD3kaNeFfY=";
+    })
+    (fetchurl {
+      inherit version;
+      url = "https://storage.googleapis.com/tensorflow/libtensorflow/libtensorflow-cpu-linux-x86_64-2.9.1.tar.gz";
+      hash = "sha256-f1ENJUbj214QsdEZRjaJAD1YeEKJKtPJW8pRz4KCAXM=";
+    })
+
+  ];
+
+  unpackPhase = ''
+    # Merge the app and the models from github
+    tar -xzpf "${builtins.elemAt srcs 0}" recognize;
+    tar -xzpf "${builtins.elemAt srcs 1}" recognize-${version}/models;
+    mv recognize-${version}/models recognize
+
+    # Place the tensorflow lib at the right place for building
+    cd recognize/node_modules/@tensorflow/tfjs-node/deps
```

Comment:
Apologies if I forgot about something we discussed before, but I'm curious, did you try to package the node_modules stuff here via Nix fetchers or is that not feasible?

(non-blocking, would like to understand the reasoning for now)

Reply:
The problem is that the install scripts of the tfjs node_modules fetches random files from the internet (in this case the src[3]) and it needs to use node-gyp. I did not find any other package that manages to fix this properly. Only photoprism has something that might be interesting that we could look into (but I would like to deffer that to a future PR)